### PR TITLE
[Avatar] Use new color system

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

This PR handles the first half of https://github.com/Shopify/polaris-ux/issues/333

I'll follow up with another PR to tackle the non latin name styles

image

Concern:
image

Dark mode colors are a little muddy and two are not passing our contrast standards. 

### WHAT is this pull request doing?

Using the new color system to match master brand styles in the Avatar component with fallbacks to the original colors.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page, Avatar, Stack} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <Stack alignment="baseline">
        <Avatar initials="KD" name="a" />
        <Avatar initials="KD" name="b" />
        <Avatar initials="KD" name="c" />
        <Avatar initials="KD" name="d" />
        <Avatar initials="KD" name="e" />
        <Avatar initials="KD" name="f" />
        <div
          style={{borderLeft: '1px solid hsl(0, 0%, 50%)', height: '40px'}}
        />
        <Avatar customer initials="KD" name="e" />
        <Avatar
          initials="JD"
          name="John D"
          source="https://images.unsplash.com/photo-1574157573248-293553e90ead?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1534&q=80"
        />
      </Stack>
      <Stack alignment="baseline">
        <Avatar
          size="small"
          initials="JD"
          name="John D"
          source="https://images.unsplash.com/photo-1574157573248-293553e90ead?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1534&q=80"
        />
        <Avatar
          initials="JD"
          name="John D"
          source="https://images.unsplash.com/photo-1574157573248-293553e90ead?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1534&q=80"
        />
        <Avatar
          size="large"
          initials="JD"
          name="John D"
          source="https://images.unsplash.com/photo-1574157573248-293553e90ead?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1534&q=80"
        />
      </Stack>
      <Stack alignment="baseline">
        <Avatar size="small" customer initials="KD" name="e" />
        <Avatar customer initials="KD" name="e" />
        <Avatar size="large" customer initials="KD" name="e" />
      </Stack>
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

File: src/utilities/theme/color-adjustments.ts [hunk 1]
```diff
@@ -590,7 +592,7 @@ export const colorAdjustments: ColorAdjustments = {
   },
   highlightText: {
     baseColor: 'highlight',
-    light: {lightness: 98},
+    light: {lightness: 38},
```
Comment (kyledurand) [inline src/utilities/theme/color-adjustments.ts]: I'm not sure about leaving this change as part of this PR but this value was well below our contrast standards.

Before | After
---|---
 |

File: src/utilities/theme/types.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ export interface ThemeConfig {
     warning?: string;
     highlight?: string;
     success?: string;
+    peach?: string;
+    cranberry?: string;
```
Comment (kyledurand) [inline src/utilities/theme/types.ts]: Peach and cranberry are what these are called in figma. We can probably think of better names for these

File: src/components/Avatar/tests/Avatar.test.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ describe('<Avatar />', () => {
     it('renders an Image component with a customer Avatar if the customer prop is true', () => {
       const avatar = mountWithAppProvider(<Avatar customer />);
       const image = avatar.find(Image);
-      expect(image.prop('source')).toContain('avatar-');
+      expect(image.prop('source')).toContain('customer-placeholder.svg');
```
Comment (alex-page) [inline src/components/Avatar/tests/Avatar.test.tsx]:
```suggestion
      expect(image.prop('source')).toBe('customer-placeholder.svg');
```

File: src/utilities/theme/color-adjustments.ts [hunk 4]
```diff
@@ -12,6 +12,8 @@ export enum UNSTABLE_Color {
   Warning = '#FFC453',
   Highlight = '#59D0C2',
   Success = '#008060',
+  Peach = '#FFC8B5',
+  Cranberry = '#FDC7CE',
```
Comment (alex-page) [inline src/utilities/theme/color-adjustments.ts]: What do you think about providing one avatar color and then changing the lightness and hue to generate the palette?

File: src/utilities/theme/color-adjustments.ts (hunk 1 above)
Comment (alex-page) [reply src/utilities/theme/color-adjustments.ts]: Looking at the banner and how this color is used I am not sure why the color is white. I might be missing some context. Best to wait for @tmlayton or @danrosenthal.

Comment (tmlayton) [reply src/utilities/theme/color-adjustments.ts]: This change seems fine, the lightness of the text for similar roles in 30-40 in light mode and 60-70 in dark mode. They should probably be consistent across the board 30/70, 40/60, or something else.

File: src/utilities/theme/color-adjustments.ts (hunk 4 above)
Comment (tmlayton) [reply src/utilities/theme/color-adjustments.ts]: I like the idea of providing a single decorative color, then perhaps adjusting based on the same hue delta as our colors to create the contrast/counter/secondary decorative color. In theory, regardless of what color is provided the resulted generated color should look nice.

That means these color roles would be relative adjustments instead of absolute ones, which is doable would just need some refactoring of the factory and new schema in the adjustments to flag this role as a relative adjustment.

Would like to talk through this a bit more though.

File: src/utilities/theme/types.ts (hunk 2 above)
Comment (tmlayton) [reply src/utilities/theme/types.ts]: Agreed, we had some early discussion around these as decorative colors. The word "decorative" is so vague, though. Any thoughts?

File: src/components/Avatar/Avatar.scss [hunk 5]
```diff
@@ -38,27 +38,28 @@ $large-size: rem(60px);
 }
 
 .styleOne {
-  background: color('teal');
+  color: var(--p-warning-text, color('white'));
```
Comment (tmlayton) [inline src/components/Avatar/Avatar.scss]: Having success and warning roles used in avatars feels wrong to me, highlight a bit less. But maybe this is a time when it is okay to have an exception to color role usage?

File: UNRELEASED.md [hunk 6]
```diff
@@ -4,6 +4,7 @@
 
 ### Enhancements
 
+- Removed customer images from avatar and updated to use new color system [#2449](https://github.com/Shopify/polaris-react/pull/2449)
```
Comment (tmlayton) [inline UNRELEASED.md]: Should customer images be removed in current Polaris, too? If that’s the case I think the customer images removal should be done in its own PR. We also have not been adding changelog entries for color system changes since they are invisible unless the feature flag is turned on and somewhat unstable.

**Comment by tmlayton** [comment]
> Dark mode colors are a little muddy and two are not passing our contrast standards.

Don’t sweat it too much, I think as is quickly changing make the best call you can, ship, and iterate.

File: src/components/Avatar/Avatar.scss (hunk 5 above)
Comment (tmlayton) [reply src/components/Avatar/Avatar.scss]: Otherwise I wonder if it’s possible, to @alex-page’s point, to take a single decorative color and generate a range of complimentary colors that actually look good regardless of the base color. I think that logic would be based on reverse engineering our use of the color.

File: src/utilities/theme/types.ts (hunk 2 above)
Comment (dleroux) [reply src/utilities/theme/types.ts]: extended, complementary? Could they be bundled under 1 role?

File: UNRELEASED.md (hunk 6 above)
Comment (kyledurand) [reply UNRELEASED.md]: I think it makes sense to remove them before the masterbrand changes as there's no way to fallback to them once the new icon is added. I'll move that to a new PR and it doesn't have to be merged right away

**Comment by kyledurand** [comment]
I've removed the changes to customer avatars and also now basing all decorative colors off of one base

image

image

Figma:
image

These all match the hsl values from the figma file nearly perfectly. Sometime values might be off by 1 but I can't seem to tell the difference.

**Comment by alex-page** [comment]
Approving the work looks great, happy to keep talking about the naming in the future!

File: src/components/Avatar/Avatar.tsx [hunk 7]
```diff
@@ -16,7 +16,7 @@ enum Status {
   Errored = 'ERRORED',
 }
 
-const STYLE_CLASSES = ['one', 'two', 'three', 'four', 'five', 'six'];
+const STYLE_CLASSES = ['one', 'two', 'three', 'four', 'five'];
```
Comment (danrosenthal) [inline src/components/Avatar/Avatar.tsx]: This will cause visual regressions, as indicated by the failing Percy test.

We can remove this class conditionally based on the `useFeatures` hook returning `globalTheming === true`, which will preserve backwards compatibility.

Comment (kyledurand) [reply src/components/Avatar/Avatar.tsx]: Oh nice, I'll make the change in a bit 👍

**Comment by kyledurand** [comment]
Fallback added for the sixth variation:
image

And looping over five with the new system:
image

It's my understanding that this PR will be put on hold until we're more confident in how we should handle decorative coloring

File: src/components/Avatar/Avatar.scss [hunk 8]
```diff
@@ -71,6 +77,7 @@ $large-size: rem(60px);
   left: 50%;
   width: 100%;
   height: 100%;
+  background-color: var(--p-neutral, color('sky'));
```
Comment (tmlayton) [inline src/components/Avatar/Avatar.scss]: `--p-surface-background`

Comment (tmlayton) [reply src/components/Avatar/Avatar.scss]: Actually maybe this should be neutral

File: src/utilities/theme/color-adjustments.ts [hunk 9]
```diff
@@ -679,4 +680,110 @@ export const colorAdjustments: ColorAdjustments = {
...
+      hue: 52,
+      saturation: 100,
+      lightness: 18,
+    },
+    dark: {lightness: 100},
+  },
+  decorativeTwoSurface: {
+    baseColor: 'decorative',
+    light: {
+      hue: 30,
```
Comment (tmlayton) [inline src/utilities/theme/color-adjustments.ts]: I think this is fine for now just to get something out there, but what I had in mind is these being relative changes in hue based on the role color.

With this approach, it ignores the base decorative color and fixes the color to Shopify’s palette. To get a relative approach working it would take updates to the typings and color factory.
